aioasuswrt/connection.py

```diff
@@ -118,7 +117,13 @@ async def async_run_command(self, command, first_try=True):
                 return []
         except TimeoutError:
             _LOGGER.error("Host timeout.")
-            return []
+            async with self._io_lock:
```

Comment:
I think it would be better to refactor this part to ensure that we call the disconnect without releasing the io_lock. Now another thread might/will get the lock before disconnecting. The retry than probably has to be done by setting a boolean in the except and calling the `async_run_command`

Reply:
You are right, fixed